lucene/misc/src/java/org/apache/lucene/misc/store/DirectIODirectory.java

```diff
@@ -0,0 +1,392 @@
+   *  buffer storage space consumed during merging. */
+
+  public final static int DEFAULT_MERGE_BUFFER_SIZE = 262144;
+
+  /** Default min expected merge size before direct IO is
+   *  used (10 MB): */
+  public final static long DEFAULT_MIN_BYTES_DIRECT = 10*1024*1024;
+
+  private final int blockSize, mergeBufferSize;
+  private final long minBytesDirect;
+  private final Path path;
+  
+  volatile boolean isOpen = true;
+
+  /** Reference to {@code com.sun.nio.file.ExtendedOpenOption.DIRECT} by reflective class and enum lookup.
+   * There are two reasons for using this instead of directly referencing ExtendedOpenOption.DIRECT:
+   * <ol>
+   * <li> ExtendedOpenOption.DIRECT is OpenJDK's internal proprietary API. This API causes un-suppressible(?) warning to be emitted
+   *  when compiling with --release flag and value N, where N is smaller than the the version of javac used for compilation.</li>
+   * <li> It is possible that Lucene is run using JDK that does not support ExtendedOpenOption.DIRECT. In such a
+   *  case, dynamic lookup allows us to bail out with UnsupportedOperationException with meaningful error message.</li>
+   * </ol>
+   * <p>This reference is {@code null}, if the JDK does not support direct I/O.
+   */
+  static final OpenOption ExtendedOpenOption_DIRECT; // visible for test
+  static {
+    OpenOption option;
+    try {
+      final Class<? extends OpenOption> clazz = Class.forName("com.sun.nio.file.ExtendedOpenOption").asSubclass(OpenOption.class);
+      option = Arrays.stream(clazz.getEnumConstants())
+                      .filter(e -> e.toString().equalsIgnoreCase("DIRECT"))
+                      .findFirst()
+                      .orElse(null);
+    } catch (Exception e) {
+      option = null;
+    }
+    ExtendedOpenOption_DIRECT = option;
+  }
+
+  /** Create a new DirectIODirectory for the named location.
+   * 
+   * @param path the path of the directory
+   * @param mergeBufferSize Size of buffer to use for
+   *    merging.
+   * @param minBytesDirect Merges, or files to be opened for
+   *   reading, smaller than this will
+   *   not use direct IO.  See {@link
+   *   #DEFAULT_MIN_BYTES_DIRECT}
+   * @param delegate fallback Directory for non-merges
+   * @throws IOException If there is a low-level I/O error
+   */
+  public DirectIODirectory(Path path, int mergeBufferSize, long minBytesDirect, FSDirectory delegate) throws IOException {
+    super(delegate);
+    this.blockSize = Math.toIntExact(Files.getFileStore(path).getBlockSize());
+    this.mergeBufferSize = mergeBufferSize;
+    this.minBytesDirect = minBytesDirect;
+    this.path = path.toRealPath();
+  }
+
+  /** Create a new DirectIODirectory for the named location.
+   * 
+   * @param path the path of the directory
+   * @param delegate fallback Directory for non-merges
+   * @throws IOException If there is a low-level I/O error
+   */
+  public DirectIODirectory(Path path, FSDirectory delegate) throws IOException {
```

Comment:
Remove path here too.

Reply:
I fixed this in https://github.com/apache/lucene-solr/pull/2052/commits/9a59b56f2f94aed53d2268c3053e0fa37382ce44